File: client-java/src/main/java/com/example/client_java/ui/TelaCadastro.java
package com.example.client_java.ui;

import com.example.client_java.model.request.CadastroRequest;
import com.example.client_java.model.response.CadastroResponse;
import com.example.client_java.service.BancoApiClient;
import org.springframework.stereotype.Component;
import org.springframework.web.client.HttpClientErrorException;

import java.util.Scanner;

@Component
public class TelaCadastro {

    private static final String[][] TIPOS = {
            {"1", "Corrente",  "Limite R$ 1.200,00",     "CORRENTE"},
            {"2", "Poupança",  "Rendimento 0,500% a.m.", "POUPANÇA"},
    };

    private final Scanner sc;
    private final BancoApiClient bancoApiClient;

    public TelaCadastro(Scanner sc, BancoApiClient bancoApiClient) {
        this.sc = sc;
        this.bancoApiClient = bancoApiClient;
    }

    public void exibir() {
        Banner.limpar();
        Banner.exibirCabecalho("ABERTURA DE CONTA");
        Banner.tituloSecao("DADOS PESSOAIS");

        String nome = lerObrigatorio("Nome completo  : ");
        if (nome == null) return;

        String cpf = lerObrigatorio("CPF            : ");
        if (cpf == null) return;

        String senha = lerSenha("Senha          : ");
        if (senha == null) return;

        String confirmaSenha = lerSenha("Confirme senha : ");
        if (confirmaSenha == null) return;

        if (!senha.equals(confirmaSenha)) {
            Banner.erro("As senhas não coincidem.");
            pausa(2000);
            return;
        }

        Banner.espaco();
        Banner.labelSecao("TIPO DE CONTA");
        Banner.espaco();
        for (String[] tipo : TIPOS) {
            String detalhe = Banner.cinza("(" + tipo[2] + ")");
            System.out.println(Banner.margem() + "  " +
                    Banner.cinza("[" + tipo[0] + "]  ") +
                    Banner.branco(String.format("%-12s", tipo[1])) + "  " + detalhe);
        }
        System.out.println(Banner.margem() + "  " + Banner.cinza("[0]  Voltar"));
        Banner.espaco();
        Banner.prompt("Tipo           :");
        String tipoStr = sc.nextLine().trim();

        if (tipoStr.equals("0")) return;
        int tipo;
        try {
            tipo = Integer.parseInt(tipoStr);
            if (tipo < 1 || tipo > 2) throw new NumberFormatException();
        } catch (NumberFormatException e) {
            Banner.erro("Tipo inválido.");
            pausa(1500);
            return;
        }

        Banner.espaco();
        Banner.linhaSeparadora();
        Banner.espaco();
        exibirResumo(nome, cpf, tipo);
        Banner.espaco();
        Banner.linhaSeparadora();
        Banner.espaco();

        System.out.println(Banner.margem() + "  " + Banner.cinza("[1]  Confirmar abertura"));
        System.out.println(Banner.margem() + "  " + Banner.cinza("[0]  Cancelar"));
        Banner.espaco();
        Banner.prompt("Escolha        :");
        String conf = sc.nextLine().trim();

        if (!conf.equals("1")) {
            Banner.info("Operação cancelada.");
            pausa(1200);
            return;
        }

        Banner.espaco();
        Banner.info("Cadastrando...");

        try {
            CadastroRequest request = new CadastroRequest(nome, cpf, senha, tipo);
            CadastroResponse resposta = bancoApiClient.cadastrar(request);

            Banner.sucesso(resposta.mensagem());
            Banner.info("Você já pode fazer login com seu CPF e senha.");
        } catch (HttpClientErrorException e) {
            String erro = e.getResponseBodyAsString();
            String mensagem = erro
                    .replaceAll(".*\"erro\":\"*", "")
                    .replaceAll("\"}.*", "")
                    .trim();

            Banner.erro("Erro ao cadastrar: " + mensagem);
            pausa(2000);
        }

        pausa(2500);
    }

    private void exibirResumo(String nome, String cpf, int tipo) {
        String[] info = TIPOS[tipo - 1];
        Banner.labelSecao("RESUMO DA ABERTURA");
        Banner.campo("Titular",    Banner.branco(nome));
        Banner.campo("CPF",        Banner.cinza(cpf));
        Banner.campo("Tipo",       tipo == 1 ? Banner.badgeAzul(info[3]) : Banner.badgeVerde(info[3]));
        Banner.campo("Condições",  Banner.cinza(info[2]));
    }

    private String lerObrigatorio(String label) {
        Banner.prompt(label);
        String valor = sc.nextLine().trim();
        if (valor.isBlank()) {
            Banner.erro("Campo obrigatório.");
            pausa(1000);
            return null;
        }
        return valor;
    }

    private String lerSenha(String label) {
        Banner.prompt(label);
        java.io.Console console = System.console();
        if (console != null) {
            char[] chars = console.readPassword();
            if (chars == null || chars.length == 0) {
                Banner.erro("Senha obrigatória.");
                pausa(1000);
                return null;
            }
            return new String(chars);
        } else {
            String valor = sc.nextLine().trim();
            if (valor.isBlank()) {
                Banner.erro("Senha obrigatória.");
                pausa(1000);
                return null;
            }
            return valor;
        }
    }

    private void pausa(long ms) {
        try { Thread.sleep(ms); } catch (InterruptedException ignored) {}
    }
}

File: client-python/ui/tela_cadastro.py
# ui/tela_cadastro.py
import time
import getpass
import sys
from ui.banner import Banner
from models import CadastroRequest

TIPOS = [
    ["1", "Corrente", "Limite R$ 1.200,00", "CORRENTE"],
    ["2", "Poupança", "Rendimento 0,500% a.m.", "POUPANÇA"],
]


def executar_cadastro(api_client):
    Banner.limpar()
    Banner.exibir_cabecalho("ABERTURA DE CONTA")
    Banner.titulo_secao("DADOS PESSOAIS")

    nome = _ler_obrigatorio("Nome completo  :")
    if nome is None: return

    cpf = _ler_obrigatorio("CPF            :")
    if cpf is None: return

    senha = _ler_senha("Senha          :")
    if senha is None: return

    confirma_senha = _ler_senha("Confirme senha :")
    if confirma_senha is None: return

    if senha != confirma_senha:
        Banner.erro("As senhas não coincidem.")
        _pausa(2000)
        return

    Banner.espaco()
    Banner.label_secao("TIPO DE CONTA")
    Banner.espaco()

    for tipo_item in TIPOS:
        detalhe = Banner.cinza(f"({tipo_item[2]})")
        print(f"{Banner.margem()}  "
              f"{Banner.cinza('[' + tipo_item[0] + ']  ')}"
              f"{Banner.branco(f'{tipo_item[1]:<12}')}  {detalhe}")

    print(f"{Banner.margem()}  {Banner.cinza('[0]  Voltar')}")
    Banner.espaco()
    Banner.prompt("Tipo            :")
    tipo_str = input().strip()

    if tipo_str == "0": return

    try:
        tipo = int(tipo_str)
        if tipo < 1 or tipo > 2:
            raise ValueError()
    except ValueError:
        Banner.erro("Tipo inválido.")
        _pausa(1500)
        return

    Banner.espaco()
    Banner.linha_separadora()
    Banner.espaco()
    _exibir_resumo(nome, cpf, tipo)
    Banner.espaco()
    Banner.linha_separadora()
    Banner.espaco()

    print(f"{Banner.margem()}  {Banner.cinza('[1]  Confirmar abertura')}")
    print(f"{Banner.margem()}  {Banner.cinza('[0]  Cancelar')}")
    Banner.espaco()
    Banner.prompt("Escolha        :")
    conf = input().strip()

    if conf != "1":
        Banner.info("Operação cancelada.")
        _pausa(1200)
        return

    Banner.espaco()
    Banner.info("Cadastrando...")
    _pausa(500)

    try:
        request = CadastroRequest(nome=nome, cpf=cpf, senha=senha, tipoConta=tipo)

        api_client.cadastrar(request)

        Banner.sucesso("Conta criada com sucesso!")
        Banner.info("Você já pode fazer login com seu CPF e senha.")
        _pausa(2500)

    except Exception as e:
        Banner.erro(f"Erro ao cadastrar: {e}")
        _pausa(2000)

def _exibir_resumo(nome: str, cpf: str, tipo: int):
    info = TIPOS[tipo - 1]
    Banner.label_secao("RESUMO DA ABERTURA")
    Banner.campo("Titular", Banner.branco(nome))
    Banner.campo("CPF", Banner.cinza(cpf))

    badge_tipo = Banner.badge_azul(info[3]) if tipo == 1 else Banner.badge_verde(info[3])
    Banner.campo("Tipo", badge_tipo)
    Banner.campo("Condições", Banner.cinza(info[2]))


def _ler_obrigatorio(label: str) -> str or None:
    Banner.prompt(label)
    valor = input().strip()
    if not valor:
        Banner.erro("Campo obrigatório.")
        _pausa(1000)
        return None
    return valor


def _ler_senha(label: str) -> str or None:
    print(f"{Banner.margem()}  {Banner.B_AZUL}{label}{Banner.RESET} ", end="")
    sys.stdout.flush()

    valor = getpass.getpass("").strip()

    if not valor:
        Banner.erro("Senha obrigatória.")
        _pausa(1000)
        return None
    return valor


def _pausa(milissegundos: int):
    time.sleep(milissegundos / 1000.0)